e2e/rules_foo/foo/repositories.bzl

```diff
@@ -0,0 +1,30 @@
+"Shows how a custom rule set can export its npm dependencies"
+
+load("@aspect_rules_js//npm:npm_import.bzl", _npm_translate_lock = "npm_translate_lock")
+
+def foo_repositories(name):
```

Comment:
@thesayyn I think we should replace the rules_ts [npm_dependencies](https://github.com/aspect-build/rules_ts/blob/797610817564c68c56ac53a80a570be02867bf21/ts/private/npm_repositories.bzl#L75-L101) with something like this.

This "repositories function" should (in most rulesets) also accept things like the `npm_translate_lock(npmrc)` so it can be configured to load from other npm repos and other npm config can be set.

@gregmagolan we should try to define a standard way of doing this just to be consistent across repos. Potentially even create a macro (wrapping `npm_translate_lock`) that rulesets like this would use just to force consistency? We should think about that before we start adding params to these methods in each repo...

Reply:
Potentially; tho the `foo_repositories()` pattern is pretty simple already so a wrapper macro wouldn't add much.

Reply:
I just worry it will slowly grow and not be simple, and if we don't think about it each ruleset will grow differently...

Reply:
Suppose there is some room to standardize on the naming of the npm_repositories.bzl.... but lets leave that for a follow-up if there is merit